packages/service/ni_measurement_plugin_sdk_service/grpc/channelpool.py

```diff
@@ -78,3 +81,32 @@ def close(self) -> None:
             for channel in self._channel_cache.values():
                 channel.close()
             self._channel_cache.clear()
+
+    def _is_local(self, target: str) -> bool:
+        hostname = ""
+        # First, check if the target string is in URL format
+        parse_result = urlparse(target)
+        if parse_result.scheme and parse_result.hostname and parse_result.port:
+            hostname = parse_result.hostname
+        else:
+            # Next, check for target string in <host_name>:<port> format
+            match = re.match(r"^(.*):(\d+)$", target)
+            if match:
+                hostname = match.group(1)
+
+        if not hostname:
+            return False
+        if hostname == "localhost" or hostname == "LOCALHOST":
+            return True
+        
+        # IPv6 addresses don't support parsing with leading/trailing brackets so we need to remove them.
+        match = re.match(r"^\[(.*)\]$", hostname)
+        if match:
+            hostname = match.group(1)
+
+        try:
+            address = ipaddress.ip_address(hostname)
+            return address.is_loopback
+        except:
+            pass
+        return False
```

Comment:
I think it's appropriate to catch ValueError here. Some of the urllib.parse functions can return TypeError when you pass the wrong data type, or UnicodeError if you pass non-ASCII characters, and I don't think we should treat those situations as local addresses.

Also, consider moving the return into the except block.

```suggestion
        except ValueError:
            return False
```